lib/ASTGen/Sources/ASTGen/ParameterClause.swift

```diff
@@ -0,0 +1,111 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import CASTBridging
+import SwiftSyntax
+
+// MARK: - ParamDecl
+
+fileprivate protocol ValueParameterSyntax: SyntaxProtocol {
+  var _firstName: TokenSyntax? { get }
```

Comment:
I’m not a fan of these underscored properties. Underscored properties to me are always things that aught to be private but aren’t for some stupid reason. 

I would prefer

```swift
/// The `firstName` with optional type. 
///
/// This is the lowest denominator between `FunctionParameterSyntax` and `EnumCaseParameterSyntax`.
var optionalFirstName: TokenSyntax? { get }
```

Reply:
These requirements are a sort of hack around our lack of support for covariant witnesses. I used underscores to highlight this, but I am happy to rename and add a FIXME instead.